Common/CostFunctions/itkAdvancedImageToImageMetric.hxx

```diff
@@ -360,6 +359,12 @@ AdvancedImageToImageMetric<TFixedImage, TMovingImage>::CheckForBSplineInterpolat
     }
     else
     {
+      using RayCastInterpolatorType =
+        itk::AdvancedRayCastInterpolateImageFunction<MovingImageType, CoordinateRepresentationType>;
+
+      assert(m_BSplineInterpolator || m_BSplineInterpolatorFloat || m_ReducedBSplineInterpolator ||
+             m_LinearInterpolator || dynamic_cast<RayCastInterpolatorType *>(interpolator));
+
       this->m_GradientImage = nullptr;
```

Comment:
@mstaring @stefanklein Do you know, what is the status of RayCastInterpolator? It isn't handled explicitly in EvaluateMovingImageValueAndDerivativeWithOptionalThreadId, meaning that it will use m_GradientImage, by default, at:

https://github.com/SuperElastix/elastix/blob/98aa401157d5608b47a3f89892e119c74b0c88fe/Common/CostFunctions/itkAdvancedImageToImageMetric.hxx#L497

However, m_GradientImage appears to be set to null, for RayCastInterpolator!